runtime/src/iree/hal/drivers/hip/hip_driver.c

```diff
@@ -465,10 +472,120 @@ static iree_status_t iree_hal_hip_driver_create_device_by_index(
   IREE_HIP_RETURN_IF_ERROR(&driver->hip_symbols,
                            hipDeviceGet(&device, device_index), "hipDeviceGet");
 
-  iree_status_t status = iree_hal_hip_driver_create_device_by_id(
-      base_driver, IREE_HIPDEVICE_TO_DEVICE_ID(device), param_count, params,
-      host_allocator, out_device);
+  *out_id = IREE_HIPDEVICE_TO_DEVICE_ID(device);
+  return iree_ok_status();
+}
 
+static iree_status_t iree_hal_hip_driver_get_device_id_by_path(
+    iree_hal_driver_t* base_driver, iree_string_view_t device_path,
+    iree_hal_device_id_t* out_id) {
+  iree_hal_hip_driver_t* driver = iree_hal_hip_driver_cast(base_driver);
+  IREE_RETURN_IF_ERROR(iree_hal_hip_init(driver));
```

Comment:
this is sketchy - try to move this to someplace deterministic and call it only there - every function shouldn't be needing to check this as it implies no one knows when hip is actually initialized (and thus a reader coming along definitely won't know where hip is actually initialized). There should be one or two places that call hip_init, they should be public API methods (or directly reachable from them), and every other method should be able to assume it's initialized.